src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts

```diff
@@ -607,54 +610,69 @@ class TerminalTabsDragAndDrop implements IListDragAndDrop<ITerminalInstance> {
 		};
 	}
 
-	drop(data: IDragAndDropData, targetInstance: ITerminalInstance | undefined, targetIndex: number | undefined, originalEvent: DragEvent): void {
+	async drop(data: IDragAndDropData, targetInstance: ITerminalInstance | undefined, targetIndex: number | undefined, originalEvent: DragEvent): Promise<void> {
 		this._autoFocusDisposable.dispose();
 		this._autoFocusInstance = undefined;
 
 		let sourceInstances: ITerminalInstance[] | undefined;
-		const terminalResources = originalEvent.dataTransfer?.getData(DataTransfers.TERMINALS);
-		if (terminalResources) {
+		const terminalResources = originalEvent.dataTransfer?.getData(DataTransfers.RESOURCES);
+		const terminals = originalEvent.dataTransfer?.getData(DataTransfers.TERMINALS);
+		if (terminals && terminalResources) {
 			const json = JSON.parse(terminalResources);
 			for (const entry of json) {
 				const uri = URI.parse(entry);
-				const instance = this._terminalService.instances.find(e => e.resource.path === uri.path);
-				if (instance) {
-					sourceInstances = [instance];
-					this._terminalService.moveToTerminalView(instance);
+				const [, workspaceId, instanceId] = uri.path.split('/');
+				if (workspaceId && instanceId) {
+					const instance = this._terminalService.instances.find(e => e.resource.path === instanceId);
+					if (instance) {
+						sourceInstances = [instance];
+						this._terminalService.moveToTerminalView(instance);
+					} else if (workspaceId !== this._workspaceContextService.getWorkspace().id) {
+						this._localTerminalService.requestDetachInstance(workspaceId, Number.parseInt(instanceId));
+						await timeout(300);
```

Comment:
this is needed in order for the terminal service to reply with the persistentProcessId